From 4337aaa69f9c12091743a71a0e6ab9350355aa16 Mon Sep 17 00:00:00 2001
From: Tomas Hlavacek <tmshlvck@gmail.com>
Date: Mon, 22 Oct 2018 16:45:51 +0200
Subject: [PATCH] Add support for multiple versions of GCC

Add support for multiple versions of GCC in the chroot build environment
based on Gentoo.
---
 Makefile | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 1b6090b..d27a4f1 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,7 @@
 ECDSA_PRIV_KEY	:= ecdsa_priv_key.txt
 CROSS_CM3	:= armv7m-none-eabi-
-CROSS_COMPILE	:= aarch64-unknown-linux-gnu-
+CROSS_ATF	:= /usr/x86_64-pc-linux-gnu/aarch64-linux-gnu/gcc-bin/5.4.0/aarch64-linux-gnu-
+CROSS_COMPILE	:= /usr/x86_64-pc-linux-gnu/aarch64-linux-gnu/gcc-bin/8.2.0/aarch64-linux-gnu-
 
 BUILD_PLAT = build/a3700/release
 
@@ -31,14 +32,14 @@ mox-imager/mox-imager:
 atf-marvell/$(BUILD_PLAT)/bl1.bin:
 	make -C atf-marvell \
 		PATH=$(ATF_GCC_PATH):$(PATH) \
-		CROSS_COMPILE=$(CROSS_COMPILE) \
+		CROSS_COMPILE=$(CROSS_ATF) \
 		DEBUG=0 LOG_LEVEL=0 USE_COHERENT_MEM=0 PLAT=a3700 \
 		$(BUILD_PLAT)/bl1.bin
 
 atf-marvell/$(BUILD_PLAT)/fip.bin: u-boot/u-boot.bin
 	make -C atf-marvell \
 		PATH=$(ATF_GCC_PATH):$(PATH) \
-		CROSS_COMPILE=$(CROSS_COMPILE) \
+		CROSS_COMPILE=$(CROSS_ATF) \
 		BL33=$(shell pwd)/u-boot/u-boot.bin DEBUG=0 LOG_LEVEL=0 USE_COHERENT_MEM=0 PLAT=a3700 \
 		$(BUILD_PLAT)/fip.bin
 
-- 
2.19.1

